File: languages/go/entity/identity/schema.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// Schema represents a schema with its attributes.
type Schema struct {
	Cid        string
	CidJsonLd  string
	SchemaType string
	Json       string
}

func NewSchemaFromProto(s *proto.Schema) Schema {
	if s == nil {
		return Schema{}
	}
	return Schema{
		Cid:        s.Cid,
		CidJsonLd:  s.CidJsonLd,
		SchemaType: s.SchemaType,
		Json:       s.Json,
	}
}

func (c Schema) ToProto() *proto.Schema {
	return &proto.Schema{
		Cid:        c.Cid,
		CidJsonLd:  c.CidJsonLd,
		SchemaType: c.SchemaType,
		Json:       c.Json,
	}
}

File: languages/js/src/entity/identity/schema.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";

/**
 * Represents a schema with its attributes.
 */
export class Schema {
  cid: string;
  cidJsonLd: string;
  schemaType: string;
  json: string;

  /**
   * Constructs a Schema object with the specified parameters.
   * @param cid 
   * @param cidJsonLd 
   * @param schemaType 
   * @param json 
   */
  constructor(
    cid: string,
    cidJsonLd: string,
    schemaType: string,
    json: string
  ) {
    this.cid = cid;
    this.cidJsonLd = cidJsonLd;
    this.schemaType = schemaType;
    this.json = json;
  }

  public toProto(): identityEntitiesProto.Schema {
    return identityEntitiesProto.Schema.fromPartial({
      cid: this.cid,
      cidJsonLd: this.cidJsonLd,
      schemaType: this.schemaType,
      json: this.json
    });
  }

  static fromProto(r: identityEntitiesProto.Schema): Schema {
    return new Schema(r.cid, r.cidJsonLd, r.schemaType, r.json);
  }
}
